keps/sig-network/4427-relaxed-dns-search-validation/README.md

```diff
@@ -0,0 +1,473 @@
+
+
+Yes, the feature can be disabled. Proposal covers the validation logic in detail,
+but briefly, existing values will be allowed with relaxed validation if the gate is disabled.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Then the relaxed validation will be allowed on new values in `dnsConfig.searches`.
+Existing values (prior to the initial roll-back) will continue to pass validation regardless
+of whether the gate is enabled or not.
+
+###### Are there any tests for feature enablement/disablement?
+
+Unit tests will cover cover the scenarios described in the Proposal section.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Since this a change to validation behind a feature gate, rollout should pose no risk.
+
+If a cluster needs to be rolled back for another reason, it's risky to enable this
+feature unless the previous version also has the flag (whether it's enabled or disabled).
+
+Since this feature allows previously invalid values in `dnsConfig.searches`, upgrading
+from a version without the gate present (i.e. before introducing this feature) and then
+enabling the gate is risky. In that scenario, if a search path is saved containing an
+underscore and then the cluster is downgraded to a previous version with no knowledge
+of the feature gate, then the downgrade may fail.
+
+See the Proposal section for recommendation on avoding this scenario.
+
+###### What specific metrics should inform a rollback?
+
+N/A
+
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+Will be tested during implementation.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+Existence of an underscore the `dnsConfig.searches` array in any pod spec or pod spec template
+would indicate the feature is in use.
+
+###### How can someone using this feature know that it is working for their instance?
+
+N/A. This is a validation change.
```

Comment:
Very minor point here, but they could write a resource with _ and see that the write succeeds.